x/profile/internal/keeper/utils_functions_test.go

```diff
@@ -0,0 +1,54 @@
+package keeper_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/desmos-labs/desmos/x/profile/internal/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestKeeper_IterateProfile(t *testing.T) {
+	creator, err := sdk.AccAddressFromBech32("cosmos1cjf97gpzwmaf30pzvaargfgr884mpp5ak8f7ns")
+	require.NoError(t, err)
+	creator2, err := sdk.AccAddressFromBech32("cosmos1y54exmx84cqtasvjnskf9f63djuuj68p7hqf47")
+	require.NoError(t, err)
+
+	creator3, err := sdk.AccAddressFromBech32("cosmos1s3nh6tafl4amaxkke9kdejhp09lk93g9ev39r4")
+	require.NoError(t, err)
+
+	creator4, err := sdk.AccAddressFromBech32("cosmos15lt0mflt6j9a9auj7yl3p20xec4xvljge0zhae")
+	require.NoError(t, err)
+
+	profiles := types.Profiles{
+		types.NewProfile("first", creator),
+		types.NewProfile("second", creator2),
+		types.NewProfile("not", creator3),
+		types.NewProfile("third", creator4),
+	}
+
+	expProfiles := types.Profiles{
+		profiles[1],
+		profiles[3],
+		profiles[0],
+	}
```

Comment:
I really don't get it why these are the expected values. Shouldn't be they `profiles[0], profiles[1]` if we stop as soon as we encounter `not` as the moniker?

Reply:
This is due to the sorting made by the actual db to the profiles.
It seems like it orders profiles based on the creator address.
I will edit the test checking other things to ensure that the profiles are iterated correctly to make the whole test more clear.